### DISCUSSION TITLE
Feature Request: allow to configure Lambda layer's

### DISCUSSION THREAD

**Discussion by cforce** [artilleryio/artillery#1790] [Artillery]
If we could configure lambda layer's on the generated Node load driver task, in my case a DataDog layer which instruments the client, i could make use of all features of trace and metrics collection,  aggregation and drilling on my APM backend. This would be a far better experience then just building on the basic json/html report.
If artillery would use a serverless yaml template to deploy its own task an extension of the task with the existing serverless framework features and extension would be a feature multiplier for artillery itself. It could even be a new 
"artillery run --platform serverless" :-)

https://docs.datadoghq.com/serverless/configuration/?tab=serverlessframework

**Comment by hassy** [comment]
hi @cforce 👋 We should be able to add support for configuring layers without bringing in the Serverless framework as a dependency. The AWS SDK which Artillery uses provides a way to do it via a call to [`updateFunctionConfiguration()`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Lambda.html#updateFunctionConfiguration-property).

I'm presuming you'd want to specify one of these layers published by Datadog: https://docs.datadoghq.com/serverless/installation/nodejs/?tab=custom. It looks like we'd have to provide the ability to override the default handler as well, as required by Datadog layer configuration. That can also be done through a call to `updateFunctionConfiguration()`.

It looks like it should be pretty straightforward, I'll add it to the roadmap.

**Reply by cforce** [reply]
I found this layer overview as well
https://github.com/DataDog/serverless-plugin-datadog/blob/master/src/layers.json
The one for the artillery driver  shall be  "nodejs16.x": "arn:aws:lambda:XXXX-1:464622532012:layer:Datadog-Node16-x:86" imho, but it might change any time if DD releases a new dd-agent.

**Reply by hassy** [reply]
Great, thank you. I created a ticket here: #1791 The user will be able to specify the ARN of any layer they want to use.
